### PR TITLE
Split unit test file

### PR DESCRIPTION
**Pull request by xsebek**
Split the unit test file into a separate file for each test tree (except miscellaneous).

It was already a thousand lines long and hard to read.

### CODE REVIEW

**Comment by xsebek** [comment]
I wanted to avoid conflicts, so I made the PR on top of `say-revival` branch.

After that is merged this should be changed to point to the main branch.

File: test/unit/TestParser.hs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                    ]
+                )
+            )
+        ]
+    , testGroup
+        "json encoding"
+        [ testCase "simple expr" (roundTrip "42 + 43")
+        , testCase "module def" (roundTrip "def x = 41 end; def y = 42 end")
+        ]
+    , testGroup
```
Comment (byorgey) [inline test/unit/TestParser.hs:154]: Not your fault, but I just realized that this test group isn't about the parser at all, but rather about type checking.  But it's nice that the same framework can deal with both.  I don't know if its worth moving them, or renaming the module?

**Review by byorgey — APPROVED** [review]

File: test/unit/TestParser.hs (hunk 1 above)
Comment (xsebek) [reply test/unit/TestParser.hs:154]: The whole test tree is named "Language - pipeline", what should I name the module and the exported test tree? :thinking: 

Ideas:
- `TestLanguagePipeline`
- `TestLanguage`
- `TestTypecheck`

Comment (byorgey) [reply test/unit/TestParser.hs:154]: Ah, sure, any of those would be good.  Personally I'd lean towards `TestLanguagePipeline`, although it's kinda long.

Comment (xsebek) [reply test/unit/TestParser.hs]: `TestLanguagePipeline` it is! :heavy_check_mark:

**Pull request merged**